lib/proxy/log.rb

```diff
@@ -12,79 +9,49 @@ def logger
   end
 
   class LoggerFactory
-    class LogFormatter < ::Logger::Formatter
-      Format = "%s, [%s%s] %5s -- %s: %s\n".freeze
-
-      def call(severity, time, progname, msg)
-        Format % [severity[0..0], format_datetime(time), request_id[0..7], severity, progname, msg2str(msg)]
-      end
-
-      def request_id
-        Thread.current.thread_variable_get(:request_id).to_s
-      end
-    end
-
     begin
       require 'syslog/logger'
-      ::Syslog::Logger.class_eval { alias_method :write, :info }
-
-      class SyslogFormatter < Syslog::Logger::Formatter
-        Format = "<%s> %s\n".freeze
-
-        def call(severity, time, progname, msg)
-          request_id.empty? ? clean(msg) : Format % [request_id[0..7], clean(msg)]
-        end
-
-        def request_id
-          Thread.current.thread_variable_get(:request_id).to_s
-        end
-      end
-
       @syslog_available = true
     rescue LoadError
-      # ignore, syslog isn't available on this platform
       @syslog_available = false
     end
 
     def self.logger
+      logger_name = 'foreman-proxy'
+      layout = Logging::Layouts.pattern(pattern: "%d %.8X{request} [%.1l] %m\n")
```

Comment:
I know that the structure is "fixed", that is the pattern is used from "day one", but shouldn't that also be configured, with defaults?

The reason for it, is that if we can use tools such as `jounald` (or syslog for that matter), there are tools that parse and display log information, and sometimes people wish to tweak them to specific format that easy to look for, and it can be useful for that.

Reply:
I am gonna make rotation configurable, this is configurable in core and it's confusing (one pattern for file loggers, one pattern for syslog/journal). Proxy now support structured output, users can integrate nicely without need of parsing files.